### PR TITLE
Modify base sequence regex

### LINKED ISSUE #221: Modify base sequence regex
Currently the specification enforces the following regular expressions to validate base sequences: `^([ACGT]+|N)$` which disallows some more complex usage of wildcards. We propose another approach, with the following regular expression: `^([ACGTN]+)$` which would allow users to find wildcards from within partly known sequences, not only as single nucleotide polymorphisms.

beaconregex
Figure 1. Validation of example base sequences according to current specification (left) and proposed form (right).

### PR DESCRIPTION
**Pull request by teemukataja**
Fixes #221 

PR to fix a forgotten issue.

### CODE REVIEW

**Comment by sdelatorrep** [comment]
Please @teemukataja, update the beacon.md so it reflects this change. Also, I think we should explain what this change means (e.g. use N as a wildcard: `ACNCCNN`) in both the beacon.md and the description of the fields in beacon.yaml.
I think we shouldn't merge this PR until this is done.

**Comment by teemukataja** [comment]
> Please @teemukataja, update the beacon.md so it reflects this change. Also, I think we should explain what this change means (e.g. use N as a wildcard: ACNCCNN) in both the beacon.md and the description of the fields in beacon.yaml.

Rebased and added explanation to description.

File: beacon.md [hunk 1]
```diff
@@ -255,15 +255,15 @@ An example `GET` request and response to the info endpoint:
 |Parameter (* mandatory, ~ one of these is mandatory)|Description|Type|Example|
 |---|---|:---:|---|
 |*referenceName**|Reference name (chromosome). Accepting values 1-22, X, Y so follows Ensembl chromosome naming convention.|string|`'1'`|
-|*referenceBases**|Reference bases for this variant (starting from `start`). Accepted values: [ACGT]* When querying for variants without specific base alterations (e.g. imprecise structural variants with separate `variantType` as well as `startMin` & `endMin` ... parameters), the use of a single "N" value is required.<br/>See the REF field in [VCF 4.2 specification](https://samtools.github.io/hts-specs/VCFv4.2.pdf).|string|`'G'`|
```
Comment (sdelatorrep) [inline beacon.md]: @teemukataja , any reason to remove the reference to the VCF specification? You kept it in the `alternateBases` description but not here.

File: beacon.md [hunk 2]
```diff
@@ -255,15 +255,15 @@ An example `GET` request and response to the info endpoint:
...
+|*referenceBases**|Reference bases for this variant (starting from `start`). Accepted values: [ACGTN]*. N is a wildcard, that denotes the position of any base, and can be used as a standalone base of any type or within a partially known sequence. For example a sequence where the first and last bases are known, but the middle portion can exhibit countless variations of [ACGT], or the bases are unknown: ANNT the Ns can take take any form of [ACGT], which makes both ACCT and ATGT (or any other combination) viable sequences.|string|`'G'`|
```
Comment (teemukataja) [inline beacon.md]:
```suggestion
|*referenceBases**|Reference bases for this variant (starting from `start`). Accepted values: [ACGTN]*. N is a wildcard, that denotes the position of any base, and can be used as a standalone base of any type or within a partially known sequence. For example a sequence where the first and last bases are known, but the middle portion can exhibit countless variations of [ACGT], or the bases are unknown: ANNT the Ns can take take any form of [ACGT], which makes both ACCT and ATGT (or any other combination) viable sequences.<br/>See the REF field in [VCF 4.2 specification](https://samtools.github.io/hts-specs/VCFv4.2.pdf).|string|`'G'`|
```

File: beacon.md (hunk 1 above)
Comment (teemukataja) [reply beacon.md]: No, must have been removed accidentally.

**Comment by mbaudis** [comment]
@sdelatorrep @teemukataja Independent of the (correct) fix here: How about patterns w/ a range of "N" bases: `/AC(?:N{1,3})TT/` ? While I don't think about providing patterns, a limited amount of alternative codings in the request could be of interest...
I've stumbled on a problem here when doing wildcard matches, where one usually would want a `*` in the `alternateBases` query; `N` returns only the single base replacements (you can test this on Beacon+ w/ using `N`, `NN` ...).

**Pull request merged**